src/components/Avatar/Avatar.Image.tsx

```diff
@@ -1,68 +1,263 @@
+// this Component leverage code from react-image for the loading part
+// https://github.com/mbrevda/react-image
+//
+// Goal would be to either only use react-image at some point, or to migrate the loading
+// code here into an <Image /> component
+
 import * as React from 'react'
 import VisuallyHidden from '../VisuallyHidden'
 import { classNames } from '../../utilities/classNames'
-import { getEasingTiming } from '../../utilities/easing'
 import { noop } from '../../utilities/other'
-import { ImageWrapperUI, ImageUI } from './styles/Avatar.css'
-
-export const AvatarImage = props => {
-  const {
-    animationDuration,
-    animationEasing,
-    className,
-    hasImage,
-    image,
-    isImageLoaded,
-    onError,
-    onLoad,
-    name,
-    title,
-  } = props
-
-  const componentClassName = classNames(
-    'c-Avatar__imageWrapper',
-    isImageLoaded && 'is-herbieFullyLoaded',
-    className
-  )
-
-  const backgroundImage = isImageLoaded ? `url('${image}')` : null
-
-  const imageStyle = {
-    transition: `opacity ${animationDuration}ms ${getEasingTiming(
-      animationEasing
-    )}`,
-  }
+import { ImageWrapperUI, ImageUI, TitleUI } from './styles/Avatar.css'
+import { getAnimationProps } from './Avatar.utils'
 
-  const contentMarkup = (
-    <ImageWrapperUI className={componentClassName} style={imageStyle}>
-      <ImageUI className="c-Avatar__image" style={{ backgroundImage }}>
-        <div className="c-Avatar__name">
-          <VisuallyHidden>{name}</VisuallyHidden>
-          <img
-            alt={name}
-            onError={onError}
-            onLoad={onLoad}
-            src={image}
-            style={{ display: 'none' }}
-          />
-        </div>
-      </ImageUI>
-    </ImageWrapperUI>
-  )
-
-  return hasImage ? contentMarkup : title
+let cache = {}
+export const clearCache = () => {
+  cache = {}
 }
+// ... 29,134,155,206
+export interface Props {
+  animation: boolean
+  animationDuration: number
+  animationEasing: string
+  className?: string
+  src?: string[] | string
+  light: boolean
+  name: string
+  onError?: () => void
+  onLoad?: () => void
+  title: string | number
+}
+
+export interface State {
+  currentIndex: number
+  isLoading: boolean
+  isLoaded: boolean
+}
```

Comment:
Minor: For consistency and keeping the component file cleaner, can we move these to a Avatar.types.ts file?

Reply:
👍 all done in 6048ade6